### ISSUE TITLE
Indent of comments

### ISSUE THREAD

**Issue by czjvic** [eslint/eslint#3845]
```
"indent": [1, 4, {
            "SwitchCase": 1,
            "VariableDeclarator": 1
        }],
```

``` javascript
function a() {
//comment
    alert('hello world')
}
```

Expected result:

```
2:1 Expected indentation of 4 space characters but found 0 indent
```

ESLint version
v1.3.1
v1.4.3
Both has same result - "no warning/no error".
##

**Comment by eslintbot** [comment]
Thanks for the issue! If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

**Comment by BYK** [comment]
Can you please answer all the questions @eslintbot asked? Most importantly, ESLint version is missing.

**Comment by czjvic** [comment]
Sorry, updated.

**Comment by BYK** [comment]
@gyandeeps does your latest indent patch fix this?

**Comment by gyandeeps** [comment]
This is not a bug as currently rule doesnt support indentation check for comments.
Its tricky also as the comments sometime belong to 2 nodes based on trailing comments or the other way around.

**Comment by BYK** [comment]
Ah, okay :)

**Comment by gyandeeps** [comment]
@eslint/eslint-team Should we add this functionality to check for comment indentation?
If yes, then how do we calculate indentation, based on `trailingComments` or `leadingComments` or combination or node specific (trailing or leading based on node type)?

**Comment by BYK** [comment]
I think we should have this functionality. Not sure if a node based approach would cover all the cases though. For instance in this example hoe would we access to the comment in the function body?

**Comment by nzakas** [comment]
I think we should.

@byk use LineComment and BlockComment in the rule to listen for comments.

**Comment by BYK** [comment]
Roger.

**Comment by xjamundx** [comment]
Can I work on this

**Comment by ilyavolodin** [comment]
@xjamundx It's been opened for almost 5 month without a progress, I think it's safe to say nobody is working on it currently:-) Would be great if you could pick it up!

**Comment by promag** [comment]
Ping.

**Comment by xjamundx** [comment]
There's been a lot of talk of re-writing indentation rule altogether, so I was holding off on that happening. There's also the tab vs. space bug which would affect comments as well. I took an hour or two on this but didn't make any great progress. Does anyone know the status of the indentation rewrite that I've heard folks talk about?

**Comment by ilyavolodin** [comment]
@BYK ping

**Comment by BYK** [comment]
Still in progress. I made some mental progress but nothing to show as a code yet.

**Comment by nzakas** [comment]
@BYK can we mark this as accepted? Is this something we want to do?

**Comment by BYK** [comment]
Yes. The rewrite does not distinguish between comments or other things, it simply enforces a certain indentation level for all lines.

**Comment by HeroicAutodidact** [comment]
Was this fixed? 

I can lint a file with the following contents:

```
(function() {
   // Empty commment
  console.log('test');
})();

if(true) {
   // foo
  console.log('bar');
 // foo
  console.log('bar');
}
```

And no errors are raised (with indentation set to 2). 

Even worse, though,  if I run `eslint --fix`, improperly indented code gets indented but leaves behind its comments.

**Comment by azhang496** [comment]
@xjamundx are you still working on this? if not i can take it on!

**Comment by HeroicAutodidact** [comment]
+I would think one would always want indentation of comments to be coupled with the node below. I can't think of any exceptions within my experience.

**Comment by nzakas** [comment]
@HeroicAutodidact issues are closed when they are fixed, so no, this is not fixed.

@azhang496 you should feel free to work on this. I don't believe anyone is working on it currently.

**Comment by azhang496** [comment]
@nzakas after looking into this, I think it's a bit out of my depth as a newcomer to the project. will defer to anyone else who wants to work on it

**Comment by nzakas** [comment]
@azhang496 no problem, thanks for giving it a shot.

**Comment by sstern6** [comment]
@nzakas would like to tackle this but having trouble reproducing the issue. 

With eslint --version `3.6.0`, as @HeroicAutodidact mentioned, not getting any errors when running @czjvic example with with the eslint args.

Could someone please confirm they are able to still reproduce this issue?

```
/* eslint "indent": [2, 4, {"SwitchCase": 1, "VariableDeclarator": 1 }] */
function a() {
//comment
    alert('hello world')
}
```
Could it be possible as a result of another PR, this issue was inherently fixed?

Thanks

**Comment by lukeapage** [comment]
@sstern6 - it should warn, not give no errors

**Comment by sstern6** [comment]
Thanks for the clarification @lukeapage.  Still not getting any `warnings` locally for the comments.

Were you able to reproduce?

**Comment by lukeapage** [comment]
The problem is that there is no warning and should be.. so no warnings means the bug is *not*  fixed.

**Comment by sstern6** [comment]
Wow @lukeapage I am so sorry, I completely misread the initial bug.  Thank you for the clarification.  Will get on it.  Hopefully will have a patch in a few days.

Apologize for the miscommunication 

Thanks

**Comment by jameswomack** [comment]
@sstern6 has a workaround or fix for this issue been auditioned yet?

**Comment by ilg-ul** [comment]
As of March 31, 2017, I still constantly get bad formatting like:

```javascript
    let config = {}
      // Default logLevel, if not found in the configuration.
      // May be redefined by the module.
    config.logLevel = 'warn'
    ModuleClass.initConfig(config)
```

while fixing the code via `standard --fix`

isn't anyone looking into this? 

I think it is a real pity that such a great tool like eslint is overshadowed by small comment issues, that linger for more than one year...

**Comment by not-an-aardvark** [comment]
This is fixed as part of the [indent rewrite](https://github.com/eslint/eslint/issues/7618), which is expected to be merged as part of the 4.0 release in a few weeks.

**Comment by ilg-ul** [comment]
great news, thank you!
